lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,335 @@
+                                .map(VcStoreItem::getUserId)
+                                .distinct()
+                                .toList();
+
+                List<ReportUserIdentityItem> reportUserIdentities =
+                        userIds.stream()
+                                .map(
+                                        usrId ->
+                                                new ReportUserIdentityItem(
+                                                        usrId, null, null, null, null))
+                                .toList();
+                reportUserIdentityScanDynamoDataStore.createOrUpdate(reportUserIdentities);
+
+                checkLambdaRemainingExecutionTime(context);
+
+                if (tableScanResult.lastEvaluatedKey() != null) {
+                    exclusiveStartKey = tableScanResult.lastEvaluatedKey();
+                    reportProcessingResult.tacticalStoreLastEvaluatedKey(exclusiveStartKey);
+                }
+            } while (exclusiveStartKey != null);
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Tactical storage scan completed to get unique users."));
+            reportProcessingResult.tacticalStoreLastEvaluatedKey(null);
+        }
+    }
+
+    private void processUsersToFindLOCAndUpdateDb(
+            ReportProcessingRequest reportProcessingRequest,
+            ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult,
+            Context context)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException,
+                    BatchDeleteException, StopBeforeLambdaTimeoutException {
+        List<ReportUserIdentityItem> totalReportUserIdentityItems = new ArrayList<>();
+        if (reportProcessingRequest.continueUserIdentityScan()) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Processing unique users to update their aggregate identities."));
+            var exclusiveStartKey = reportProcessingRequest.userIdentitylastEvaluatedKey();
+
+            do {
+                TableScanResult<ReportUserIdentityItem> tableScanResult =
+                        reportUserIdentityScanDynamoDataStore.getItems(
+                                exclusiveStartKey, ATTR_NAME_USER_ID, "identity", "migrated");
+
+                var userIds =
+                        tableScanResult.items().stream()
+                                .map(ReportUserIdentityItem::getUserId)
+                                .toList();
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Checking user identity for total (%s) users.",
+                                        userIds.size())));
+                List<ReportUserIdentityItem> reportUserIdentityItems = new ArrayList<>();
```

Comment:
This is still trying to store a record for every user in the table, which is going to be huge - do we actually need this for anything?